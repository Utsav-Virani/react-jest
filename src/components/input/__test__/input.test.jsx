import React from "react";
import ReactDOM from "react-dom";
import { render, cleanup , fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Input from "../input";
import Button from "../../button/button";

afterEach(cleanup);

test("should render", () => {
    const div = document.createElement("input");
    ReactDOM.render(<Input />, div);
});

test("matches snapshot", () => {
    const tree = renderer.create(<Input />).toJSON();
    expect(tree).toMatchSnapshot();
});


// test("matches snapshot", () => {
//     const input = render(<Input />);
//     fireEvent.change(input,{target: {value:"React Jest Testing..."}});
//     // fireEvent.click(<Button label="Clicl!"/>);
//     const { getByText } = render(<Input />);
//     // const { getByText } = render(<Button label="Click!"/>);
//     getByText("React Jest Testing...");
//     // getByText("Click!");
    
// });