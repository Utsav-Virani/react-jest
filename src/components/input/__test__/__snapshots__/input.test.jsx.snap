// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <label
    htmlFor="name"
  >
    Enter Name:-
  </label>,
  <input
    className="input-name-style"
    data-testid="input"
    id="name"
    name="name"
    type="text"
    value=""
  />,
]
`;
